### ISSUE TITLE
TokenAware Policy not hitting primary node

### ISSUE THREAD

**Issue by qrpike** [gocql/gocql#917]
I have set my cluster host policy to use token aware with round robin as the backup. 

Code:
```golang
	// Connect to Cassandra Cluster:
	cassandraHosts := strings.Split(cfg.CassandraHosts, ",")
	fmt.Println("Cassandra Hosts:", cfg, cassandraHosts)
	cluster := gocql.NewCluster( cassandraHosts... )
	cluster.Keyspace 			= "keyspace"
	cluster.ConnectTimeout 		= (time.Second*8)
	cluster.Timeout 			= (time.Second*8)
	cluster.ProtoVersion 		= 4
	cluster.Consistency 		= gocql.One
	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(gocql.RoundRobinHostPolicy())
	session, _ := cluster.CreateSession()
	defer session.Close()
```

However usually the Query.Latency() is ~150ms, sometimes it's 2-300ms. I enabled tracing and seems in the tracing logs it's still proxying requests through nodes.

```Tracing session b76ab55045f411e79f3403e5aa4aa17a (coordinator: 10.2.90.10, duration: 5.375ms):
2017/05/31 11:31:40.965: reading data from /10.2.106.25 (source: 10.2.90.10, elapsed: 430)
2017/05/31 11:31:40.965: Sending READ message to /10.2.106.25 (source: 10.2.90.10, elapsed: 667)
2017/05/31 11:31:40.969: REQUEST_RESPONSE message received from /10.2.106.25 (source: 10.2.90.10, elapsed: 4778)
2017/05/31 11:31:40.97: Processing response from /10.2.106.25 (source: 10.2.90.10, elapsed: 4888)
2017/05/31 11:31:40.971: READ message received from /10.2.90.10 (source: 10.2.106.25, elapsed: 16)
2017/05/31 11:31:40.971: Executing single-partition query on trades (source: 10.2.106.25, elapsed: 208)
2017/05/31 11:31:40.971: Acquiring sstable references (source: 10.2.106.25, elapsed: 285)
2017/05/31 11:31:40.971: Bloom filter allows skipping sstable 99 (source: 10.2.106.25, elapsed: 459)
2017/05/31 11:31:40.971: Bloom filter allows skipping sstable 97 (source: 10.2.106.25, elapsed: 530)
2017/05/31 11:31:40.971: Bloom filter allows skipping sstable 92 (source: 10.2.106.25, elapsed: 588)
2017/05/31 11:31:40.972: Bloom filter allows skipping sstable 83 (source: 10.2.106.25, elapsed: 659)
2017/05/31 11:31:40.972: Bloom filter allows skipping sstable 74 (source: 10.2.106.25, elapsed: 740)
2017/05/31 11:31:40.972: Key cache hit for sstable 37 (source: 10.2.106.25, elapsed: 859)
2017/05/31 11:31:40.972: Skipped 0/6 non-slice-intersecting sstables, included 0 due to tombstones (source: 10.2.106.25, elapsed: 1033)
2017/05/31 11:31:40.974: Merged data from memtables and 6 sstables (source: 10.2.106.25, elapsed: 2860)
2017/05/31 11:31:40.974: Read 100 live and 0 tombstone cells (source: 10.2.106.25, elapsed: 2951)
2017/05/31 11:31:40.974: Enqueuing response to /10.2.90.10 (source: 10.2.106.25, elapsed: 3023)
2017/05/31 11:31:40.974: Sending REQUEST_RESPONSE message to node-1.cassandra/10.2.90.10 (source: 10.2.106.25, elapsed: 3126)
```

I'm somewhat new to Cassandra but this line`reading data from /10.2.106.25 (source: 10.2.90.10,` makes me thing it's proxing through the 10.2.90.10 node to get to the 10.2.106.25 node.

Any insight would be greatly appreciated. And maybe I'm just reading this wrong.

Thanks,

**Comment by Zariel** [comment]
What does the query look like? Is it using bind variables?

**Comment by Zariel** [comment]
This should be fixed via #1035
